chain-crypto/src/algorithms/sumed25519/mod.rs

```diff
@@ -120,23 +120,27 @@ impl KeyEvolvingSignatureAlgorithm for SumEd25519_12 {
 mod tests {
     use super::*;
 
-    #[quickcheck]
-    /// `public_from_binary`should fail if the provided byte array does not match the public key size
-    fn public_from_binary_size_check(n: usize) {
+    use proptest::prelude::*;
+    use test_strategy::proptest;
+    // `public_from_binary`should fail if the provided byte array does not match the public key size
+    // the size is limited to 1 MiB to avoid segfaults during testing
+    #[proptest]
+    fn public_from_binary_size_check(#[strategy(..1_048_576usize)] n: usize) {
```

Comment:
Here

Reply:
It's overkill for this and similar tests, though: most of the times it will test extravagantly large sizes with no difference to the branch coverage at all.
A more effective strategy would be biased, selecting alternatively from larger, smaller, and the correct size.

In fact, this is a waste of proptest, it could be just a few fixed test cases for the correct size, one-off cases, empty slice, and a large size.